### PR TITLE
`Position.TraceIndex`: panic when `maxDepth < p.depth` happens

### PR DESCRIPTION
**Pull request by zhiqiangxu**
Currently when this happens, the error is being caught outside like [here](https://github.com/ethereum-optimism/optimism/blob/6583735983ce39f66dba544cdfbae83e7f9c780c/op-challenger/game/fault/trace/outputs/provider.go#L54).

But this is error prone and the wrong result may be returned unnoticed, it's better to error out more explicitly. Considering the positions this function is being called and to make the change small, this PR makes it panic when `maxDepth < p.depth` happens.

### CODE REVIEW

File: op-challenger/game/fault/types/position.go [hunk 1]
```diff
@@ -82,9 +82,13 @@ func (p Position) lshIndex(amount Depth) *big.Int {
 
 // TraceIndex calculates the what the index of the claim value would be inside the trace.
 // It is equivalent to going right until the final depth has been reached.
+// Note: this method will panic if maxDepth < p.depth
 func (p Position) TraceIndex(maxDepth Depth) *big.Int {
 	// When we go right, we do a shift left and set the bottom bit to be 1.
 	// To do this in a single step, do all the shifts at once & or in all 1s for the bottom bits.
+	if maxDepth < p.depth {
+		panic("maxDepth < p.depth")
```
Comment (tynes) [inline op-challenger/game/fault/types/position.go]: Perhaps `fmt.Sprintf` with the values that caused the panic itself

Comment (zhiqiangxu) [reply op-challenger/game/fault/types/position.go]: Good idea! Done.

**Review by tynes — APPROVED** [review]

**Comment by tynes** [comment]
If you rebase then it should fix the `contracts-bedrock-coverage` issue and this is good to merge

**Comment by zhiqiangxu** [comment]
It seems something is wrong with CI itself?

**Comment by tynes** [comment]
This needs a rebase then should be good to merge, its failing due to something that we reverted

**Pull request merged**